tests/integration/components/file-details/dynamic-scan/manual-test.js

```
@@ -0,0 +1,1222 @@
+      this.server.get('/profiles/:id/device_preference', (schema, req) => {
+        return {
+          ...schema.devicePreferences.find(`${req.params.id}`)?.toJSON(),
+          device_type: ENUMS.DEVICE_TYPE.TABLET_REQUIRED,
+        };
+      });
+
+      this.server.put('/profiles/:id/device_preference', (_, req) => {
+        const data = req.requestBody
+          .split('&')
+          .map((it) => it.split('='))
+          .reduce((acc, [key, value]) => ({ ...acc, [key]: value }), {});
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)